Products/EEAContentTypes/browser/language.py

```diff
@@ -138,7 +151,11 @@ def data(self):
                 translation = translations[code][0]
 
                 if putils.isDefaultPage(translation):
-                    translation = aq_parent(aq_inner(translation))
+                    # Check if parent container has translation and
+                    # in this case redirect to it.
+                    my_container = aq_parent(aq_inner(translation))
+                    if my_container.getTranslations().has_key(code):
+                        translation = aq_parent(aq_inner(translation))
```

Comment:
you could set translation = my_container as the logic is already calculated at 156

Reply:
done.